### PR TITLE
Add a quickstart section

### PR DESCRIPTION
**Pull request by OlgaMaciaszek**

### CODE REVIEW

**Review by spencergibb — COMMENTED** [review]
At a glance I didn't see spring config import mentioned. Is also remove the reference to the all starter

**Comment by OlgaMaciaszek** [comment]
> At a glance I didn't see spring config import mentioned.

It's not there cause I will add it to master only.

**Comment by spencergibb** [comment]
Sorry, didn't notice this was targeting 2.2.x

**Review by spencergibb — APPROVED** [review]

File: README.adoc [hunk 1]
```diff
@@ -50,6 +50,232 @@ with Spring Cloud Netflix provides Intelligent Routing (Zuul), Client Side Load
 :sc-ext: java
 :project-full-name: Spring Cloud Zookeeper
 
+== Quick Start
+
+This quick start walks through using Spring Cloud Zookeeper for Service Discovery and Distributed Configuration.
+
+First, run Zookeeper on your machine, then you can access it and use as Service Registry and Configuration source with Spring Cloud Zookeeper.
```
Comment (Buzzardo) [inline README.adoc:57]: Change to:

First, run Zookeeper on your machine. Then you can access it and use it as a Service Registry and Configuration source with Spring Cloud Zookeeper.

File: README.adoc [hunk 2]
```diff
@@ -50,6 +50,232 @@ with Spring Cloud Netflix provides Intelligent Routing (Zuul), Client Side Load
...
+
+=== Discovery Client Usage
+
+To use these features in an application, you can build it as a Spring Boot application that depends on `spring-cloud-zookeeper-core` and `spring-cloud-zookeeper-discovery`.
+The most convenient way to add the dependency is with a Spring Boot starter `org.springframework.cloud:spring-cloud-starter-zookeeper-discovery`.
```
Comment (Buzzardo) [inline README.adoc:62]: Add a colon (:) after "starter".

File: README.adoc [hunk 3]
```diff
@@ -50,6 +50,232 @@ with Spring Cloud Netflix provides Intelligent Routing (Zuul), Client Side Load
...
+      <plugin>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-maven-plugin</artifactId>
+      </plugin>
+    </plugins>
+  </build>
+</project>
+----
+
+The example below shows a typical Gradle setup:
```
Comment (Buzzardo) [inline README.adoc:110]: Change "The example below" to "The following example".

File: README.adoc [hunk 4]
```diff
@@ -50,6 +50,232 @@ with Spring Cloud Netflix provides Intelligent Routing (Zuul), Client Side Load
...
+    }
+
+    public static void main(String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+
+}
+----
+
+When this HTTP server runs, it connects to Zookeeper running at the default local 2181 port.
```
Comment (Buzzardo) [inline README.adoc:155]: Change to:

When this HTTP server runs, it connects to Zookeeper, which runs on the default local port (2181).

File: README.adoc [hunk 5]
```diff
@@ -50,6 +50,232 @@ with Spring Cloud Netflix provides Intelligent Routing (Zuul), Client Side Load
...
+To modify the startup behavior, you can change the location of Zookeeper by using `application.properties` as shown in the following example:
```
Comment (Buzzardo) [inline README.adoc:156]: Add a comma before "as shown".

File: README.adoc [hunk 6]
```diff
@@ -50,6 +50,232 @@ with Spring Cloud Netflix provides Intelligent Routing (Zuul), Client Side Load
...
+
+----
+spring:
+  cloud:
+    zookeeper:
+      connect-string: localhost:2181
+----
+
+You can now use `DiscoveryClient`, `@LoadBalanced RestTemplate` or `@LoadBalanced WebClient.Builder` to retrieve services and instances data from Zookeeper, as shown in the following example:
```
Comment (Buzzardo) [inline README.adoc:165]: Add a comma after "`@LoadBalanced RestTemplate`".

File: README.adoc [hunk 7]
```diff
@@ -50,6 +50,232 @@ with Spring Cloud Netflix provides Intelligent Routing (Zuul), Client Side Load
...
+      <plugin>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-maven-plugin</artifactId>
+      </plugin>
+    </plugins>
+  </build>
+</project>
+----
+
+The example below shows a typical Gradle setup:
```
Comment (Buzzardo) [inline README.adoc:232]: Change "The example below" to "The following example".

File: README.adoc [hunk 8]
```diff
@@ -50,6 +50,232 @@ with Spring Cloud Netflix provides Intelligent Routing (Zuul), Client Side Load
...
+    }
+
+    public static void main(String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+
+}
+----
+
+The application will retrieve configuration data from Zookeeper.
```
Comment (Buzzardo) [inline README.adoc:277]: Change "will retrieve" to "retrieves".

File: docs/src/main/asciidoc/quickstart.adoc [hunk 9]
```diff
@@ -0,0 +1,223 @@
+This quick start walks through using Spring Cloud Zookeeper for Service Discovery and Distributed Configuration.
+
+First, run Zookeeper on your machine, then you can access it and use as Service Registry and Configuration source with Spring Cloud Zookeeper.
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/quickstart.adoc:3]: Change to:

First, run Zookeeper on your machine. Then you can access it and use it as a Service Registry and Configuration source with Spring Cloud Zookeeper.

File: docs/src/main/asciidoc/quickstart.adoc [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+
+=== Discovery Client Usage
+
+To use these features in an application, you can build it as a Spring Boot application that depends on `spring-cloud-zookeeper-core` and `spring-cloud-zookeeper-discovery`.
+The most convenient way to add the dependency is with a Spring Boot starter `org.springframework.cloud:spring-cloud-starter-zookeeper-discovery`.
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/quickstart.adoc:8]: Add a colon (:) after "starter".

File: docs/src/main/asciidoc/quickstart.adoc [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+      <plugin>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-maven-plugin</artifactId>
+      </plugin>
+    </plugins>
+  </build>
+</project>
+----
+
+The example below shows a typical Gradle setup:
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/quickstart.adoc:56]: Change "The example below" to "The following example".

File: docs/src/main/asciidoc/quickstart.adoc [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+    }
+
+    public static void main(String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+
+}
+----
+
+When this HTTP server runs, it connects to Zookeeper running at the default local 2181 port.
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/quickstart.adoc:101]: Change to:

When this HTTP server runs, it connects to Zookeeper, which runs on the default local port (2181).

File: docs/src/main/asciidoc/quickstart.adoc [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+To modify the startup behavior, you can change the location of Zookeeper by using `application.properties` as shown in the following example:
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/quickstart.adoc:102]: Add a comma before "as shown".

File: docs/src/main/asciidoc/quickstart.adoc [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+
+----
+spring:
+  cloud:
+    zookeeper:
+      connect-string: localhost:2181
+----
+
+You can now use `DiscoveryClient`, `@LoadBalanced RestTemplate` or `@LoadBalanced WebClient.Builder` to retrieve services and instances data from Zookeeper, as shown in the following example:
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/quickstart.adoc:111]: Add a comma after "`@LoadBalanced RestTemplate`".

File: docs/src/main/asciidoc/quickstart.adoc [hunk 15]
```diff
@@ -0,0 +1,223 @@
...
+        return list.get(0).getUri().toString();
+    }
+    return null;
+}
+----
+
+=== Distributed Configuration Usage
+
+To use these features in an application, you can build it as a Spring Boot application that depends on `spring-cloud-zookeeper-core` and `spring-cloud-zookeeper-config`.
+The most convenient way to add the dependency is with a Spring Boot starter `org.springframework.cloud:spring-cloud-starter-zookeeper-config`.
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/quickstart.adoc:130]: Add a colon (:) after "starter".

File: docs/src/main/asciidoc/quickstart.adoc [hunk 16]
```diff
@@ -0,0 +1,223 @@
...
+      <plugin>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-maven-plugin</artifactId>
+      </plugin>
+    </plugins>
+  </build>
+</project>
+----
+
+The example below shows a typical Gradle setup:
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/quickstart.adoc:178]: Change "The example below" to "The following example".

**Review by Buzzardo — CHANGES_REQUESTED** [review]
The README.adoc file at the top of the project can include docs/src/main/asciidoc/quickstart.adoc, so that you don't have duplicate content.

**Comment by OlgaMaciaszek** [comment]
> The README.adoc file at the top of the project can include docs/src/main/asciidoc/quickstart.adoc, so that you don't have duplicate content.

This one is automatically generate from the other one that contains the `include` directive.

**Comment by Buzzardo** [comment]
> > The README.adoc file at the top of the project can include docs/src/main/asciidoc/quickstart.adoc, so that you don't have duplicate content.
> 
> This one is automatically generate from the other one that contains the `include` directive.

Aha. OK. I didn't need to edit it, then. No harm done, though. :)

**Comment by OlgaMaciaszek** [comment]
Sorry for this. It's basically just the same text that gets automatically embeded.

**Review by Buzzardo — APPROVED** [review]

**Pull request merged**